docs/DesignDocs/system_ocall_opt_in.md

```diff
@@ -0,0 +1,152 @@
+# Explicit Enclave Opt-in to System OCalls
+
+System OCalls are OCalls that are part of Open Enclave core libraries. These are generally not
+called by the enclave developer but instead are called by other functions internal to OE SDK.
+
+Because OCalls have potential security impact, an enclave developer should be able to opt-out
+of these OCalls being made and should have visibility into what OCalls might be called by OE
+SDK.
+
+## EDL installation
+
+In order to support ocall opt-in, system EDL files will now be installed as part of the SDK at
+`${install_path}/include/openenclave/edl/`. This allows them to be imported into user edl by the
+enclave developer.
+
+## Developer opt-in
+
+In order to use any system OCall, it must be imported into the enclave's EDL file. If it is not
+included, but the OCall is still called by the enclave application or some part of the SDK,
+enclave linking will fail.
+
+Users can import a system OCall the same way any other edl file would be imported. For example,
+if an enclave with edl file `sample.edl` wants to use the function `oe_log_ocall()` they can do
+one of the following:
+
+*Note: These examples assume that `${install_path}/include` is passed to edger8r as an edl search path*
+
+1. Import just the required OCall
+
+```
+// sample.edl
+enclave
+{
+    from "openenclave/edl/logging.edl" import oe_log_ocall;
+
+    trusted
+    {
+        public void my_sample_ecall();
+    };
+
+    untrusted
+    {
+        public void my_sample_ocall();
+    };
+}
+```
+
+2. Import all OCalls from `tee.edl`
+
+```
+// sample.edl
+enclave
+{
+    from "openenclave/edl/logging.edl" import *;
+
+    trusted
+    {
+        public void my_sample_ecall();
+    };
+
+    untrusted
+    {
+        public void my_sample_ocall();
+    };
+}
+```
+
+In example (1), linking would fail if the enclave made a call to `oe_realloc_ocall()`
+(also part of tee.edl) unless the developer modified the import line to
+`from "openenclave/edl/tee.edl" import oe_log_ocall, oe_realloc_ocall;`. In this scenario,
+example (2) will work without modification.
+
+## EDL local structures
+
+Under this design, system edl files will prefer using local structures; that is, structures
+that are defined in edl. This is distinct from foreign structures, which are defined in
+C/C++ header files and included in the edl file.
+
+In the case that a foreign structure must be used, system edl files may only include headers
+that are installed with the SDK.
+
+Local EDL structures provide additional security benefits detailed [here](
+https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/full_edl_serialization.md#motivation)
+so foreign structures should be avoided whenever possible.
+
+## Developer opt-out
```

Comment:
@CodeMonkeyLeet and @jbdelcuv I have updated this section to make it more clear what it means to be able to opt-out of using these OCalls. Let me know if this makes sense